### DISCUSSION TITLE
Plugin name schema should be a reference to "plugins"

### DISCUSSION THREAD

**Discussion by enstyled** [Kong/kong#7343] [Ideas and feature requests]
### Summary

Since the plugin `name` should be one of the plugins installed in the cluster, it will probably be better if instead of type "string" it is a reference to `plugins`, similar to `routes`:

```
{
    "fields": [
        ...
        {
            "name": {
                "required": true,
                "type": "string"
            }
        },
        ...
        {
            "route": {
                "reference": "routes",
                "on_delete": "cascade",
                "default": null,
                "type": "foreign"
            }
        },
        ...
    ]
}
```

### Steps To Reproduce

1. Open `[YOUR_KONG_ADMIN_URL]/schemas/plugins`

### Additional Details & Logs

- I am building an Admin UI that uses the schemas to dynamically generate the create/edit forms for the different entities.
- Kong version (2.3.3)

**Comment by Tieske** [comment]
@enstyled fyi the Kong manager GUI is available in a free version these days, no need to build your own.

**Comment by enstyled** [comment]
@Tieske Interesting! Can you point me to any documentation about it? The official one says Kong Manager is a part of Kong Enterprise.

**Comment by Tieske** [comment]
I think you can just follow Kong Enterprise documentation
(without a license the enterprise version only adds the GUI, not the other options, but it remains a closed source version)
